### ISSUE TITLE
Too many false positives

### ISSUE THREAD

**Issue by kmiloarguello** [crownedone/mushrooms#2]
Hello, @kevinzagalo 
To determine whether a mushroom is edible or not, we use a KNN algorithm after applying a dimensional reduction using PCA. However, for the evaluation part. It seems the we got a lot of false positives.

How do we can reduce that number ?

https://github.com/crownedone/mushrooms/commit/43322e9ea92700e42b68e691ff68b8df221589cc#diff-2e45a60ba3f23302f85463ce5c9745e1R1493

**Comment by kevinzagalo** [comment]
You should try PCA with no dimension reduction and reduce it progressively. Are you sure that your 2 component PCA is optimal ?

What results do you have without PCA ?

Check https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html> and show me your precision/recall for each class

**Comment by kmiloarguello** [comment]
These are the results without PCA but taking in account the feature scaling. 

![image](https://user-images.githubusercontent.com/13356409/79755515-8c54a500-8319-11ea-989b-1edd1ffed635.png)

Our precision increased. Here how they look like. In the top without PCA and at the bottom with PCA.

![image](https://user-images.githubusercontent.com/13356409/79759138-97f69a80-831e-11ea-9bea-e8d327c0bd70.png)

**Comment by kevinzagalo** [comment]
How many components did you use ?

It would be interesting to see how you select that parameter ;) -  the n_component parameter- , in the same way we chose the degree of the polynomial transformation here : https://framagit.org/kzagalo/IMC-4302C/-/blob/master/Lab3/Lab3_Model_Selection.ipynb

It looks better without PCA. Is there a big improvement in computation times ? Maybe you could compare the execution times to convince me that using PCA is better (or not) ?

**Comment by kmiloarguello** [comment]
Hello prof. Kevin, 

Sorry for my late answer, you know (the exams). You were right, our `n_component=2` didn't fulfill our expectations. Based on your example and with some modifications due to problems with python, we were able to find the best value of `n_component` based on this graph. We measure the accuracy over the components to find a good value. Initially we had 2 which has a bad accuracy. Now our model is at least an accuracy of `97%`.

![image](https://user-images.githubusercontent.com/13356409/80706765-24f0df00-8ae9-11ea-9f41-f92402bd4ac0.png)
